src/Microsoft.OData.Edm/ExtensionMethods/AnnotationsTargetHelper.cs

```diff
@@ -0,0 +1,205 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="AnnotationsTargetHelper.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Edm
+{
+    using System.Collections.Generic;
+    using Microsoft.OData.Edm.Vocabularies;
+
+    /// <summary>
+    /// AnnotationsTarget helper
+    /// </summary>
+    internal static class AnnotationsTargetHelper
+    {
+        public static IEdmVocabularyAnnotatable HandleTargetSegments(IEdmModel model, List<IEdmElement> results, string[] targetSegments, int targetSegmentsCount, int level)

```

Comment:
>HandleTargetSegments [](http://example.com/codeflow?start=48&length=20)

Should add UT for each public static method.

Reply:
I have added tests for the extension method. The other helper methods are private.